### ISSUE TITLE
Accessing an out-of-bounds index in bytes (calldata) does not cause an execution error.

### ISSUE THREAD

**Issue by Subway2023** [matter-labs/era-compiler-solidity#270]
```solidity
contract C {
    function fromSlice(bytes calldata c) external returns (bytes8) {
        return bytes8(c[1:9]);
    }
}
```
c=0x2c99

## solc+evm
revert
## zksolc+era_test_node
The execution is correct, but the return value is undefined. Why doesn't it cause an execution error like solc+evm?
